vc-verifier/kotlin/vcverifier/src/main/java/io/mosip/vercred/vcverifier/credentialverifier/types/msomdoc/MsoMdocVerifiableCredential.kt

```
@@ -59,27 +122,77 @@ class MsoMdocVerifiableCredential : VerifiableCredential {
+    }
+
+    private fun validateMsoPayload(issuerAuth: Array, isLatest: Boolean) {
+        try {
+            val payload = CborDecoder.decode((issuerAuth[2] as ByteString).bytes)[0]
+            if (isLatest && payload.majorType != MajorType.BYTE_STRING || payload.tag != Tag(24)) {
+                throw ValidationException("mso is not tagged", ERROR_CODE_INVALID_MSO)
+            }
+
+        } catch (exception: ValidationException) {
+            throw exception
+        } catch (exception: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)